apis/v1alpha1/gateway_types.go

```diff
@@ -125,20 +125,27 @@ type GatewaySpec struct {
 // combination of Hostname, Port, and Protocol. This will be enforced by a
 // validating webhook.
 type Listener struct {
-	// Hostname specifies to match the virtual hostname for
-	// protocol types that define this concept.
+	// Hostname specifies the virtual hostname to match for protocol types that
+	// define this concept. When unspecified, all hostnames are matched.
 	//
-	// Incoming requests that include a hostname are matched
-	// according to the given HostnameMatchType to select
-	// the Routes from this Listener.
+	// Hostname is the fully qualified domain name of a
+	// network host, as defined by RFC 3986. Note the following deviations from
+	// the "host" part of the URI as defined in the RFC:
 	//
-	// If a match type other than "Any" is supplied, it MUST
-	// be compatible with the specified Protocol field.
+	// 1. IPs are not allowed.
+	// 2. The `:` delimiter is not respected because ports are not allowed.
+	//
+	// Hostname can be "precise" which is a domain name without the terminating
+	// dot of a network host (e.g. "foo.example.com") or "wildcard", which is a
+	// domain name prefixed with a single wildcard label (e.g. "*.example.com").
+	// The wildcard character '*' must appear by itself as the first DNS label
+	// and matches only a single label.
```

Comment:
What does a hostname of`*` mean?

Reply:
Good call, I've updated the spec to clarify that this is equivalent to the hostname field being empty or unspecified.

Reply:
IIRC,`*` is explicitly not allowed in HTTPRoute?

Reply:
Yeah, I'm not sure why we made that distinction there. It is really not necessary here either, but since our minLength is 1, we can't prevent `*` from being set. It seemed safest just to treat that as one would expect, identical to the empty case. HTTPRoute is a bit different because we do have a list of hostnames, but I'm not sure that we need to explicitly prevent `*` from being set there. For reference, the comment on HTTPRoute states:

```
You cannot have a wildcard label by itself (e.g. Host == "*")
```